### ISSUE TITLE
WaveFileWriter writes samples as zeroes when using 32bits per sample

### ISSUE THREAD

**Issue by 210342** [naudio/NAudio#651]
Assuming that the samples are in [0, 1] range, the following line will almost always write a 0 to the stream, resulting in a silent .wav file:
https://github.com/naudio/NAudio/blob/1f2cbc729c3cdbce2229edb1e0d617e9af35b555/NAudio/Wave/WaveOutputs/WaveFileWriter.cs#L261

**Comment by markheath** [comment]
you're right  - that looks like a bug to me. Generally I don't recommend using that method anymore since it's quite an inefficient way to create a wav file

**Comment by markheath** [comment]
To be honest, those `WriteSample` methods are very old and looking at them several of the cases behave very oddly. Will have to revisit these. Generally I recommend people just use WaveFileWriter.CreateWaveFile or WaveFileWriter.CreateWaveFile16.
